Fix Nav bar padding & edge2edge integration

yusufnasserdev opened the pull request:
# What does this PR do?

- Improved Edge-to-Edge Content Handling:
  - On screens where the app's navigation bar is hidden (like details pages), dynamic bottom padding is now applied to footer content.

  - This is done by adding `WindowInsets.navigationBars.asPaddingValues().calculateBottomPadding()`, ensuring that the footer actions aren't not obscured by the system bottom bar.

- Navigation Bar Visibility Bug Fix:
  - Fixed a bug where the bottom navigation bar was incorrectly displayed on the Actor Details screen.
  
  - The visibility logic was corrected by adding `currentDestination?.hasRoute<ActorDetails>() != true` to the condition, ensuring the bar is hidden on the correct screens.

- Enabled True Transparent Navigation:

  - To support these changes, `Window.setNavigationBarContrastEnforced(false)` is set in MainActivity. This is a foundational step that prevents the Android system from drawing a protective scrim, allowing our app to take full control of the navigation space as recommended by the [Android Design Guidelines](https://developer.android.com/design/ui/mobile/guides/foundations/system-bars).

- Animated Navigation Bar Visibility:
  - The bottom navigation bar now animates in and out using AnimatedVisibility during screen transitions. This creates a more fluid and professional user experience, replacing the previous abrupt show/hide behavior.


## Type of Change
- 🐛 Bug fix
- 🎨 UI changes

## Changes Made
- Compose UI

## Testing
- Tested locally

## Checklist
- [x] Self-reviewed
- [x] Tests pass
- [x] Ready for review

yusufnasserdev committed:
refactor: hoist NovixApp system nav bars padding to the scaffold

yusufnasserdev committed:
fix: revert app nav bar component padding to previous state

yusufnasserdev committed:
merge branch develop into fix/nav-bar-padding

yusufnasserdev committed:
fix: enforce no navigation bar contrast

yusufnasserdev committed:
Merge branch 'develop' into fix/nav-bar-padding

yusufnasserdev committed:
fix: adjust FooterSection padding for navigation bar

yusufnasserdev committed:
feat: animate bottom navigation bar visibility

yusufnasserdev committed:
fix: hide bottom navigation bar on actor details screen

ahmeddsamii commented on app/src/main/java/com/london/app/MainActivity.kt:
```diff
@@ -1,18 +1,23 @@
         enableEdgeToEdge()
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            window.setNavigationBarContrastEnforced(false)
```
as we are in kotlin we don't need to use setter or getter like this like in java, just use isNavigationBarContrastEnforced = false

ahmeddsamii commented on app/src/main/java/com/london/app/navigation/NovixApp.kt:
```diff
@@ -165,8 +179,7 @@ fun NovixApp() {
-                    }
-                    , onNavigateToCast = { actorId->
+                    }, onNavigateToCast = { actorId ->
                         navController.navigate(Screen.ActorDetails(actorId))
```
screen here is redundant

ahmeddsamii commented on app/src/main/java/com/london/app/navigation/NovixApp.kt:
```diff
@@ -65,16 +70,25 @@ fun NovixApp() {
+            AnimatedVisibility(
+                visible = showBottomNav,
+                enter = slideInVertically(
+                    animationSpec = tween(300, easing = FastOutSlowInEasing),
```
i think you can delete 300 because the default value is 300

ahmeddsamii commented on app/src/main/java/com/london/app/navigation/NovixApp.kt:
```diff
@@ -65,16 +70,25 @@ fun NovixApp() {
+                    initialOffsetY = { it }
+                ),
+                exit = slideOutVertically(
+                    animationSpec = tween(300, easing = FastOutLinearInEasing),
```
same here

ahmeddsamii commented on presentation/src/main/java/com/london/presentation/composables/FooterSection.kt:
```diff
@@ -36,17 +39,20 @@ fun FooterSection(
-            )
+        PrimaryButton(
+            text = null,
+            onClick = { onStarClick() },
```
just type onClick = onStarClick we don't need to use curly braces and rounded also

london-bot[bot] committed:
Branch was auto-updated.

london-bot[bot] committed:
Branch was auto-updated.

Mostafa-alsaygh commented on designSystem/src/main/java/com/london/designsystem/component/AppNavBar.kt:
```diff
@@ -79,7 +79,8 @@ fun <T> NavBar(
             .topBorder(navBarColors.topBorderColor, 1.dp)
-            .background(color = navBarColors.backgroundColor),
+            .background(color = navBarColors.backgroundColor)
+            .padding(vertical = 7.dp),
```
odd padding is not right

yusufnasserdev replied:
This is following the design for this component and how it's implemented with respect to the blurred ellipse.

london-bot[bot] committed:
Branch was auto-updated.

london-bot[bot] committed:
Branch was auto-updated.

yusufnasserdev committed:
fix: bottom navigation visibility and navigation to actor details

The bottom navigation bar is now only visible on the main screens (Home, Search, Categories, Bookmarks, Account).

The navigation to actor details has been corrected to use the `ActorDetails` route.

Additionally, the animation easing for the bottom navigation visibility transition has been simplified.

yusufnasserdev committed:
fix: Use property access for `isNavigationBarContrastEnforced`

The method `setNavigationBarContrastEnforced` is deprecated. This commit updates the code to use the equivalent property access `isNavigationBarContrastEnforced`.

yusufnasserdev committed:
fix: remove unnecessary lambda in FooterSection onClick

london-bot[bot] committed:
Branch was auto-updated.

marwanqashwa reviewed (approved):
Great job

ahmeddsamii reviewed (approved):
Good job

Commit:
Merge branch 'develop' into fix/edge2edge
